src/clj/puppetlabs/services/jruby_pool_manager/jruby_core.clj

```diff
@@ -162,21 +164,39 @@
           (integer? compat-version) double
           :true str))
 
+(schema/defn ^:always-validate warn-if-compat-version-setting-not-supported
+  [parsed-compat-version :- schema/Str]
+  (let [const-compat-version (jruby-internal/get-compat-version
+                              parsed-compat-version)
+        instance-config (RubyInstanceConfig.)]
+    (.setCompatVersion instance-config const-compat-version)
+    (when-not (= const-compat-version (.getCompatVersion instance-config))
+      (log/warnf
+       "%s %s %s"
+       (i18n/trs "compat-version is set to `{0}`, which is not a supported version."
+                 parsed-compat-version)
+       (i18n/trs "Version `{0}` will be used instead." Constants/RUBY_VERSION)
+       (i18n/trs "Setting 'compat-version' is deprecated and will be removed in a future release.")))))
+
 (schema/defn ^:always-validate
   initialize-config :- jruby-schemas/JRubyConfig
   "Initialize keys with default settings if they are not given a value.
   The config is validated after these defaults are set."
-  [config :- {schema/Keyword schema/Any}]
-  (-> config
-      (update-in [:compile-mode] #(keyword (or % default-jruby-compile-mode)))
-      (update-in [:compat-version] #(parse-compat-version (or % default-jruby-compat-version)))
-      (update-in [:borrow-timeout] #(or % default-borrow-timeout))
-      (update-in [:flush-timeout] #(or % default-flush-timeout))
-      (update-in [:max-active-instances] #(or % (default-pool-size (ks/num-cpus))))
-      (update-in [:max-borrows-per-instance] #(or % 0))
-      (update-in [:environment-vars] #(or % {}))
-      (update-in [:lifecycle] initialize-lifecycle-fns)
-      jruby-internal/initialize-gem-path))
+  [{:keys [compat-version] :as config} :- {schema/Keyword schema/Any}]
+  (let [parsed-compat-version (parse-compat-version
+                               (or compat-version default-jruby-compat-version))]
+    (when compat-version
+      (warn-if-compat-version-setting-not-supported parsed-compat-version))
```

Comment:
It's a little bit weird that in this case (`compat-version` is set and we log a warning and use JRuby 9k's built in version) we ultimately set `compat-version` below too, especially since this is then logged as part of the JRuby config during startup.